src/Uno.UI/UI/Xaml/Automation/AutomationProperties.cs

```diff
@@ -148,6 +149,13 @@ private static void OnAutomationIdChanged(DependencyObject dependencyObject, Dep
 			if (FrameworkElementHelper.IsUiAutomationMappingEnabled && dependencyObject is UIElement uiElement)
 			{
 				uiElement.SetAttribute("xamlautomationid", (string)args.NewValue);
+				uiElement.SetAttribute("aria-label", (string)args.NewValue);
+
+				var role = GetHtmlRole(uiElement);
+				if (!string.IsNullOrWhiteSpace(role))
```

Comment:
TIP: As it's only internal well-known values, the `GetHtmlRole` should return `null` (and actually should be renamed to `**Find**HtmlRole`) and here the comparison should only check for `null`, it would be much faster. You could also use `string.Empty` and only compare with it if you prefer, but probably a bit less performant. I think it's not a critical path here so it's does not really matter, but just a better practice IMHO.

Reply:
Updated the PR with your suggestions. They were good ones.